completion/tests/test_services.py

```diff
@@ -205,6 +205,20 @@ def test_blocks_to_mark_complete_on_view(self):
             []
         )
 
+    def test_matches_vertical_optional_completion(self):
+        for optional_vertical, optional_child, should_include_child in (
+            (True, True, True),
+            (True, False, True),
+            (False, False, True),
+            (False, True, False),  # do not count optional children for non-optional verticals
+        ):
```

Comment:
Nit: we should use `@ddt.data` for this.

Reply:
done